AdobeStockClient/Test/Unit/Model/SearchParametersProvider/WordsTest.php

```diff
@@ -0,0 +1,339 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace AdobeStockClient\Test\Unit\Model\SearchParametersProvider;
+
+use AdobeStock\Api\Exception\StockApi;
+use AdobeStock\Api\Models\SearchParameters;
+use Magento\AdobeStockClient\Model\SearchParametersProvider\Words;
+use Magento\Framework\Api\Filter;
+use Magento\Framework\Api\Search\FilterGroup;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Escaper;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test that symbols " and \ will be removed from request.
+ */
+class WordsTest extends TestCase
```

Comment:
@IvanPletnyov @engcom-Golf please use data provider for different inputs and expected output

Reply:
@sivaschenko dataProvider was added